process/sync/metaForkDetector.go

```diff
@@ -79,7 +79,13 @@ func (mfd *metaForkDetector) AddHeader(
 	return nil
 }
 
-func (mfd *metaForkDetector) checkMetaBlockValidity(header data.HeaderHandler) error {
+func (mfd *metaForkDetector) checkMetaBlockValidity(header data.HeaderHandler, state process.BlockHeaderState) error {
```

Comment:
maybe change the name to something like shouldConsiderBlockInForkDetector, as the block might be valid, but we don't want to accept it because of the nonce

same for shard fork detector

Reply:
Done